synapse/lib/ast.py

```diff
@@ -3300,149 +3372,136 @@ async def feedfunc():
 
-                    else:
-                        assert issub
-                        # we cant remove something we cant norm...
-                        # but that also means it can't be in the array so...
-                        for v in valu:
-                            norm, info = prop.type.arraytype.norm(v)
-                            try:
-                                arry.remove(norm)
-                            except ValueError:
-                                pass
-
-                    valu = arry
-
-                if isinstance(prop.type, s_types.Ival):
-                    oldv = node.get(name)
-                    if oldv is not None:
-                        valu, _ = prop.type.norm(valu)
-                        valu = prop.type.merge(oldv, valu)
-
-                await node.set(name, valu)
-
-            except excignore:
-                pass
+            if isinstance(prop.type, s_types.Ival):
+                oldv = node.get(name)
+                if oldv is not None:
+                    valu, _ = prop.type.norm(valu)
+                    valu = prop.type.merge(oldv, valu)
 
-            yield node, path
+            await node.set(name, valu)
 
-            await asyncio.sleep(0)
+        except self.excignore:
+            pass
+
+        return node, path
 
 class EditPropDel(Edit):
 
-    async def run(self, runt, genr):
+    async def runEdit(self, runt, node, path):
 
-        if runt.readonly:
-            mesg = 'Storm runtime is in readonly mode, cannot create or edit nodes and other graph data.'
-            raise s_exc.IsReadOnly(mesg=mesg)
+        name = await self.kids[0].compute(runt, path)
 
-        async for node, path in genr:
-            name = await self.kids[0].compute(runt, path)
+        prop = node.form.props.get(name)
+        if prop is None:
+            mesg = f'No property named {name}.'
+            raise s_exc.NoSuchProp(mesg=mesg, name=name, form=node.form.name)
 
-            prop = node.form.props.get(name)
-            if prop is None:
-                mesg = f'No property named {name}.'
-                raise s_exc.NoSuchProp(mesg=mesg, name=name, form=node.form.name)
+        runt.layerConfirm(('node', 'prop', 'del', prop.full))
 
-            runt.layerConfirm(('node', 'prop', 'del', prop.full))
+        await node.pop(name)
 
-            await node.pop(name)
+        return node, path
 
-            yield node, path
+class EditUnivDel(Edit):
 
-            await asyncio.sleep(0)
+    def prepare(self):
 
-class EditUnivDel(Edit):
+        self.univprop = self.kids[0]
+        assert isinstance(self.univprop, UnivProp)
 
-    async def run(self, runt, genr):
+        if self.univprop.isconst:
+            self.validname = None
 
-        if runt.readonly:
-            mesg = 'Storm runtime is in readonly mode, cannot create or edit nodes and other graph data.'
-            raise s_exc.IsReadOnly(mesg=mesg)
+    async def runEdit(self, runt, node, path):
```

Comment:
I like where you cut the abstraction here. runEdit() as an individual node operation is very clear.